Dshop/apps/products_catalogue/tests/test_price_change_history.py

```diff
@@ -0,0 +1,139 @@
+"""
++ there is PriceChangeHistory model with relation to product, with fields: price, created_at, disabled_at.
+- creating product creates first PriceChangeHistory object with disabled_at == null
+- every price change on product creates new PriceChangeHistory object with current date and time, previous PriceChangeHistory has disabled_at set to current date and time. 
+- current price of the product is equal to the most recent entry in PriceChangeHistory for this model
+
+"""
+
+import pytest
+from freezegun import freeze_time
+from apps.products_catalogue.models import PriceChangeHistory, Product, Category
+from decimal import Decimal
+from pytest import approx
+
+
+# use https://pypi.org/project/freezegun/ for time management in tests
+# add __init__ method to product and store current price in local variable
+# in save check if price changed
+# if price changed 
+#   - update last existing PriceCHangeHistory object with disabled_at = current time
+#   - create new PriceCHangeHistory with new price
+
+
+@pytest.mark.django_db # to be deleted
+def test_model_structure():
+    # This kind of test is for fun mostly and will be deleted in going further. 
+    # However, those simple tests have great unlocking power and allow to learn new stuff. 
+    # There is nothing wrong in deleting obsolete test at some point:
+    # Flow:
+    # red
+    # green
+    # refactor
+
+    # Given
+    fields_to_check = ["product", "price", "created_at", "disabled_at"]
+    # When
+    model = PriceChangeHistory
+    # Then
+    for field in fields_to_check:
+        assert model._meta.get_field(field)
+
+
+@pytest.fixture
+def create_product_with_cat():
```

Comment:
```suggestion
def create_product_with_category():
```
full names are more readable.